### DISCUSSION TITLE
Jotai behaves differently using object instead of array

### DISCUSSION THREAD

**Discussion by subzero79** [pmndrs/jotai#1581] [Q&A]
## Summary
I was trying to refactor my current global app notification implementation using zustand to use jotai. The current setup involves a an object with randomly generated keys and the notification object inside it. 

Something like this

```ts
interface INot {
	message: string | JSX.Element;
	title: string;
        clearCb?: ()=>void
	duration: number | null;
	status: "error" | "success";
}

// Then the state of the object is
const notifications: <Partial<Record<string, INot>>> = {}
```

The keys are generated uuid's 

Then the notifications are rendered using Object.keys, i could've used a list but i choose an object i cannot recall why I did it like that.

In my first attempt to use the same with object keys i could not get the list to render. But when i switched to an array it started working.

Is this expected?

I left the codesandbox to reproduce this

## Link to reproduction

[Codesandbox](https://codesandbox.io/p/sandbox/awesome-paper-vivy4z?file=%2Fpages%2Findex.tsx&selection=%5B%7B%22endColumn%22%3A35%2C%22endLineNumber%22%3A53%2C%22startColumn%22%3A35%2C%22startLineNumber%22%3A53%7D%5D&workspace=%257B%2522activeFileId%2522%253A%2522cl8skx9ay000hlpge4cua49bz%2522%252C%2522openFiles%2522%253A%255B%2522%252Fpages%252Findex.tsx%2522%252C%2522%252Fatoms%252Findex.ts%2522%255D%252C%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522gitSidebarPanel%2522%253A%2522COMMIT%2522%252C%2522sidekickItems%2522%253A%255B%257B%2522type%2522%253A%2522PREVIEW%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522key%2522%253A%2522clakvh1qf01j52v6sifyyk4bh%2522%252C%2522isMinimized%2522%253Afalse%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522key%2522%253A%2522clakvgzk401hm2v6sl4oxltex%2522%252C%2522isMinimized%2522%253Afalse%257D%255D%257D)

## Check List

Please do not ask questions in issues.

- [ ] I understand this is not a question.

Please include a minimal reproduction.

- [ ] I've added a link to a [typescript playground](https://www.typescriptlang.org/play) or [codesandbox](https://codesandbox.io) with a minimal reproduction.

**Comment by dai-shi** [comment]
It doesn't sound jotai specific issue. probably it's reproducible with useState.

Can you create a _minimal_ example with CRA codesandbox? 👉  https://react.ts.new

**Comment by subzero79** [comment]
This is the one with CRA [codesandbox](https://codesandbox.io/s/gallant-gould-08xw0c)

**Reply by dai-shi** [reply]
https://codesandbox.io/s/white-cdn-7smd9l
Looks like you already know how to fix in comments.

**Reply by subzero79** [reply]
ahh yes...thank you I see my mistake now. Thanks

**Comment by subzero79** [comment]
I'll see if i can reproduce with setState

**Reply by subzero79** [reply]
Added setState example to the codesanbox.
